build.gradle

```diff
@@ -211,13 +232,32 @@ dependencies {
 
     configurations.all {
         resolutionStrategy {
-            force("com.google.guava:guava:33.4.8-jre") // CVE for 31.1, keep to force transitive dependencies
+            force("com.google.guava:guava:${versions.guava}")
+
+            // Jackson dependencies
+            force("com.fasterxml.jackson.core:jackson-core:${versions.jackson}")
```

Comment:
Just curious why do we need to force all of these libraries. What other version are they getting? Also from where?

Reply:
OpenSearch Java client brings in 2.17.0.

DDB Client brings in 2.18.2.

They come in via the DDB client (depending ultimately on AOS client) via jackson BOM.
```
com.fasterxml.jackson.core:jackson-core:2.18.2 FAILED
+--- com.fasterxml.jackson:jackson-bom:2.18.2
|    +--- testRuntimeClasspath
|    +--- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.18.2
|    |    +--- testRuntimeClasspath
|    |    +--- org.opensearch:opensearch-remote-metadata-sdk-ddb-client:3.1.0.0-SNAPSHOT:20250507.054903-3
```

Reply:
what about other libraries like: `opensearch-rest-client`, `jakarta`, `yasson`, `parsson`, `json`, `httpclient5`. Do we need to force versions for all of them?

Reply:
> what about other libraries ... Do we need to force versions for all of them?

All of these come from the `opensearch-java` client (which is 2.x compatible) and conflict with `opensearch` 3.x.  Until we have a 3.x version of the java client we have conflicts.

Options are to either force or exclude from the transitive bringing them in, if we are certain we're getting them.

The problem with excluding is that we only bring them in here as test dependencies (for integ tests which use clients) so we'd need to upgrade those to runtime dependencies (how they are used in SDK).

We could force to 3.x versions at SDK Client, but that means that upstream client would be responsible for maintaining everyone's versions, and I don't want to assume that extra management burden.

Reply:
Possible long-term solutions:
1. For jackson, see if using the version catalog `aws` dependency resolves it at SDK level: https://github.com/opensearch-project/OpenSearch/blob/3118fb621e82c6c8ed036ef3a83ea93946c1a0fb/gradle/libs.versions.toml#L56
2. For OpenSearch Java Client, re-evaluate when there is a 3.x version of the client

Reply:
> For jackson, see if using the version catalog aws dependency resolves it at SDK level

That's a good change to make but it's still the same `opensearch-java` issue and the same "exclude and upgrade test dependencies to runtime" vs. "force version" tradeoff.  I'll add a comment on them to make it more clear and track revisiting it later.

Reply:
Looks like OpenSearch Java v3 is out. Let me try to see if that resolves all this.